civihr_employee_portal/src/Forms/OnboardingWebForm.php

```diff
@@ -111,4 +111,72 @@ private function setWorkEmailAsPrimary($node, $values, $contactID) {
     civicrm_api3('Email', 'create', $params);
   }
 
+  /**
+   * Handles all alterations from hook_form_alter().
+   *
+   * @param array $form
+   */
+  public function alter(&$form) {
+    $this->removeEmptyKeys($form);
+    $this->addHelpText($form);
+  }
+
+  /**
+   * Remove empty components keys from $form['submitted'] as they break markup.
+   * @see https://www.drupal.org/node/2916491
+   *
+   * @param array $form
+   */
+  private function removeEmptyKeys(&$form) {
+    $form['submitted'] = array_filter(\CRM_Utils_Array::value('submitted', $form));
+  }
+
+  /**
+   * Adds help text to inform existing users of the system why they're being
+   * asked to complete the onboarding form.
+   *
+   * @param array $form
+   */
+  private function addHelpText(&$form) {
+    $helpText = 'CiviHR users can now complete a quick and easy'
+      . ' wizard to enter their details into the system.<br/>Any information '
+      . 'that you have already provided to the system will be shown in the '
+      . 'wizard and can be updated.';
+
+    $currentPage = $form['progressbar']['#page_num'];
+    $isFirstPage = $currentPage === 1;
+
+    if (!$isFirstPage || $this->userCreatedAfterOnboardingReleased()) {
+      return;
+    }
+
+    // create a 'markup' element to show message
+    $progressBarWeight = $form['progressbar']['#weight'];
+    $classes = 'alert alert-success';
+    $style = 'display: inline-block';
+    $format = '<p class="%s" style="%s">%s</p>';
+    $markup = sprintf($format, $classes, $style, $helpText);
+
+    $form['submitted']['onboarding_explanation'] = [
+      '#weight' => $progressBarWeight + 1,
+      '#type' => 'markup',
+      '#markup' => $markup,
+      '#prefix' => '<div style="text-align: center;">',
+      '#suffix' => '</div>'
+    ];
+  }
+
+  /**
+   * Checks if the current logged in user was created after the onboarding
+   * feature was released.
+   *
+   * @return bool
+   */
+  private function userCreatedAfterOnboardingReleased() {
+    global $user;
+    $onboardingRelease = (new \DateTime('13 November 2017'))->getTimestamp();
```

Comment:
This should not be done for multiple reasons:
- No business logic should be based on a release process
- Releases can be delayed
- Clients sites don't get updated on the same date as the new version is released. Also, not all client sites are updated on the same day. What happens if I created a user after the release but before my site gets updated?

If we need display different things based on if the user was created via the onboarding wizard or some other way, then we need to store this information somewhere. The first thing that comes to my mind is adding a flag to the database.